server/controllers/bookController.js

```diff
@@ -71,68 +58,92 @@ class BookController {
-
-      if (booksNotReturned.count >= 3) {
-        return Utils.errorStatus(res, 400, 'You cannot have more than 3 books in your possession');        
-      } 
-      const today = new Date();
-      const overdueBook = booksNotReturned.rows.filter((book) => book.dueDate < today);
-  
-      if(overdueBook.length >= 1) {
-        const dueBooks = overdueBook.map((book) =>`The book titled '${book.title.toUpperCase()}' was due for return on ${book.dueDate.toString().substring(0, 15)},`);
-        const dueBookTwo = dueBooks[1] ? dueBooks[1] : '' ;
-        const responseObj = {
-          message: `${dueBooks[0]} ${dueBookTwo} Kindly return before we can proceed with a new request`
-        }
-        return Utils.errorStatus(res, 400, responseObj); 
+    const booksNotReturned = await models.BorrowedBooks.findAndCountAll({ where: { patronId, returned : false } });
+
+    if (booksNotReturned.count >= 3) {
+      return Utils.errorStatus(res, 400, 'You cannot have more than 3 books in your possession');        
+    } 
+    const today = new Date();
+    const overdueBook = booksNotReturned.rows.filter((book) => book.dueDate < today);
+
+    if(overdueBook.length >= 1) {
+      const dueBooks = overdueBook.map((book) =>`The book titled '${book.title.toUpperCase()}' was due for return on ${book.dueDate.toString().substring(0, 15)},`);
+      const dueBookTwo = dueBooks[1] ? dueBooks[1] : '' ;
+      const responseObj = {
+        message: `${dueBooks[0]} ${dueBookTwo} Kindly return before we can proceed with a new request`
       }
+      return Utils.errorStatus(res, 400, responseObj); 
+    }
 
-      const dateBorrowed = new Date();
-      const dueDate = new Date();
-      // Books are due for return after 3 days
-      dueDate.setDate(dueDate.getDate() + 3); 
-      const bookDetails = await models.Books.findOne({ where: { isbn }});
-      const { title } = bookDetails.dataValues ;
-      const borrowedBook = await models.BorrowedBooks.create({
-        isbn,
-        title,
-        dateBorrowed,
-        dueDate,
-        patronId    
-      });
+    const dateBorrowed = new Date();
+    const dueDate = new Date();
+    // Books are due for return after 3 days
+    dueDate.setDate(dueDate.getDate() + 3); 
+    const bookDetails = await models.Books.findOne({ where: { isbn }});
+    const { title } = bookDetails.dataValues ;
+    const borrowedBook = await models.BorrowedBooks.create({
+      isbn,
+      title,
+      dateBorrowed,
+      dueDate,
+      patronId    
+    });
+
+    await models.Books.update({ status: 'borrowed' }, { where: { isbn } });
+    return Utils.successStatus(res, 201, 'Success', borrowedBook);
+  }
 
-      await models.Books.update({ status: 'borrowed' }, { where: { isbn } });
+  static async extendDueDate(req, res) {
+    const { id } = req.params;
+    const { date } = req.body;
+    const book = await models.BorrowedBooks.findByPk(id);
```

Comment:
I'm wondering why you are checking the book with the ID and not the ISBN? I would suggest you use the ISBN instead of the book's id.

Reply:
As for this, I think we should discuss this as a team. Maybe at the EPIC tower tomorrow. Cuz personally, I think it would be better to have like a borrowing ID for each book given to our patrons because an ID would be unique and the ISBN even tho unique to books may not really be unique in our library. Except if we plan to have only one of each book and that's very unlikely for a library. So let's discuss this and I'll make the change as soon as we agree. Thanks. @encodedBicoding @ocranbillions @cvjude @Hadeneekeh @PelumiAlesh @OvieMudi

Reply:
ISBN are unique numbers. Just like ID. and ISBN are specifically for books

Reply:
What if we have 50 copies of "Harry Potter, The Deathly Hallows"? They're all going to have the same ISBN. How can we keep track of the particular one held by a patron?

Reply:
I agree with @chuxmykel's point. The ID is unique to an individual book irrespective of the title, unlike the ISBN that will only be unique to a book based on its title.